plugins/plugin-traefik/plugin-traefik-docker/src/main/java/org/eclipse/che/plugin/traefik/TraefikCreateContainerInterceptor.java

```diff
@@ -0,0 +1,148 @@
+/*******************************************************************************
+ * Copyright (c) 2012-2017 Codenvy, S.A.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Codenvy, S.A. - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.che.plugin.traefik;
+
+import com.google.common.collect.ImmutableSet;
+import com.google.inject.name.Named;
+
+import org.aopalliance.intercept.MethodInterceptor;
+import org.aopalliance.intercept.MethodInvocation;
+import org.eclipse.che.commons.annotation.Nullable;
+import org.eclipse.che.plugin.docker.client.DockerConnector;
+import org.eclipse.che.plugin.docker.client.json.ContainerConfig;
+import org.eclipse.che.plugin.docker.client.json.ImageInfo;
+import org.eclipse.che.plugin.docker.client.params.CreateContainerParams;
+import org.eclipse.che.plugin.docker.client.params.InspectImageParams;
+import org.eclipse.che.plugin.docker.machine.CustomServerEvaluationStrategy;
+import org.eclipse.che.plugin.docker.machine.ServerEvaluationStrategy;
+import org.eclipse.che.plugin.docker.machine.ServerEvaluationStrategyProvider;
+
+import javax.inject.Inject;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Stream;
+
+import static java.lang.String.format;
+
+/**
+ * Intercept the calls on createContainer on docker Connector.
+ *
+ * @author Florent Benoit
+ */
+public class TraefikCreateContainerInterceptor implements MethodInterceptor {
+
+    /**
+     * Inject the server evaluation strategy provider.
+     */
+    private ServerEvaluationStrategyProvider serverEvaluationStrategyProvider;
+
+    /**
+     * Template.
+     */
+    private String template;
+
+
+    /**
+     * Grab labels of the config and from image to get all exposed ports and the labels defined if any
+     *
+     * @param methodInvocation
+     *         intercepting data of createContainer method on {@link DockerConnector}
+     * @return the result of the intercepted method
+     * @throws Throwable
+     *         if there is an exception
+     */
+    @Override
+    public Object invoke(MethodInvocation methodInvocation) throws Throwable {
+        ServerEvaluationStrategy serverEvaluationStrategy = serverEvaluationStrategyProvider.get();
+        // Abort if custom server evaluation strategy is not enabled.
+        if (!(CustomServerEvaluationStrategy.class.isInstance(serverEvaluationStrategy))) {
```

Comment:
Correct me if I'm mistaken but with this code, the interceptor is enabled whenever the custom strategy is applied.
If so what if the custom strategy is used with some external proxy instead of Traefik.
Then the custom strategy looks rather Traefik oriented than custom while from the configuration point of view it looks like it really custom and can be used in a wide range of situations.